server/src/main/java/org/elasticsearch/common/lucene/CommitPoint.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.lucene;
+
+import org.apache.lucene.index.IndexCommit;
+import org.apache.lucene.index.SegmentInfos;
+import org.apache.lucene.store.Directory;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+
+public final class CommitPoint extends IndexCommit {
```

Comment:
Hmm do we need a whole `CommitPoint` here? I would expect just a `SegmentInfos` to be enough for our purposes.

Reply:
It's just to get the directory that wrote the commit (to take into account multiple data paths)

Reply:
Ah ok, let's just use our own `record` to tuple them together here then.

Reply:
In fact, since we're getting the segment infos via `SegmentInfos.readLatestCommit(directory)` after the commit completes, can we just return the `Directory`? Or even return nothing from the commit and then allow the caller to read the latest commit as another step?